src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c

```diff
@@ -117,52 +147,55 @@ void func_808DE660(EnHorseLinkChild* this) {
     }
 }
 
-f32 func_808DE728(EnHorseLinkChild* this) {
-    f32 phi_fv1;
+f32 EnHorseLinkChild_GetAnimSpeed(EnHorseLinkChild* this) {
+    f32 animSpeed;
 
-    if (this->unk_148 == 2) {
-        phi_fv1 = D_808DFF18[this->unk_148] * this->actor.speed * (1.0f / 2.0f);
-    } else if (this->unk_148 == 3) {
-        phi_fv1 = D_808DFF18[this->unk_148] * this->actor.speed * (1.0f / 3.0f);
-    } else if (this->unk_148 == 4) {
-        phi_fv1 = D_808DFF18[this->unk_148] * this->actor.speed * (1.0f / 5.0f);
+    if (this->animIndex == OOT_EPONA_ANIMATION_WALK) {
+        animSpeed = sAnimPlaySpeeds[this->animIndex] * this->actor.speed * (1.0f / 2.0f);
+    } else if (this->animIndex == OOT_EPONA_ANIMATION_TROT) {
+        animSpeed = sAnimPlaySpeeds[this->animIndex] * this->actor.speed * (1.0f / 3.0f);
+    } else if (this->animIndex == OOT_EPONA_ANIMATION_GALLOP) {
+        animSpeed = sAnimPlaySpeeds[this->animIndex] * this->actor.speed * (1.0f / 5.0f);
     } else {
-        phi_fv1 = D_808DFF18[this->unk_148];
+        animSpeed = sAnimPlaySpeeds[this->animIndex];
     }
-    return phi_fv1;
+
+    return animSpeed;
 }
 
 void EnHorseLinkChild_Init(Actor* thisx, PlayState* play) {
     EnHorseLinkChild* this = THIS;
 
     Actor_ProcessInitChain(&this->actor, sInitChain);
-    Actor_SetScale(&this->actor, 0.00648f);
+    Actor_SetScale(&this->actor, 64.8 * 0.0001);
```

Comment:
Would adding the `f` at the end of both values match?
```suggestion
    Actor_SetScale(&this->actor, 64.8f * 0.0001f);
```